### ISSUE TITLE
Check that the tool's docstring and and the tool's type hints (if it has them) match

### ISSUE THREAD

**Issue by acu192** [Rhobota/lasagna-ai#10]
We should use `inspect.getfullargspec()` to get the arguments of a function and compare it with the parsed docstring to ensure:
 - match list of params
 - match type hints (if present)
 - match optional with having default values
 - match enum with list of type literals

This will help remove possible bugs when they don't match.

**Comment by acu192** [comment]
Also! We need to support multi-line `:param:` descriptions.

E.g. Add these tests (and them make them pass)

```python
...

    desc, params = parse_docstring("""
        Hi this is the docs.
        :param: thing: float: (optional) the thing
        and a second line for the first param
        :param: another: enum a b other: another value
        and a second line for the second param
    """)
    assert desc == "Hi this is the docs."
    assert params == [
        {
            'name': 'thing',
            'type': 'float',
            'description': '(optional) the thing\nand a second line for the first param',
        },
        {
            'name': 'another',
            'type': 'enum a b other',
            'description': 'another value\nand a second line for the second param',
        },
    ]

    desc, params = parse_docstring("""
        Hi this is the docs.

        :param: thing: float: (optional) the thing
        and a second line for the first param

        :param: another: enum a b other: another value
        and a second line for the second param
    """)
    assert desc == "Hi this is the docs."
    assert params == [
        {
            'name': 'thing',
            'type': 'float',
            'description': '(optional) the thing\nand a second line for the first param',
        },
        {
            'name': 'another',
            'type': 'enum a b other',
            'description': 'another value\nand a second line for the second param',
        },
    ]

...
```

**Comment by acu192** [comment]
Fixed in #35
